core/domain/story_fetchers_test.py

```diff
@@ -118,13 +181,53 @@ def test_get_stories_by_ids_for_non_existing_story_returns_none(self):
 
     def test_get_story_summary_by_id(self):
         story_summary = story_fetchers.get_story_summary_by_id(self.STORY_ID)
-
         self.assertEqual(story_summary.id, self.STORY_ID)
         self.assertEqual(story_summary.title, 'Title')
         self.assertEqual(story_summary.description, 'Description')
         self.assertEqual(story_summary.node_titles, ['Title 1'])
         self.assertEqual(story_summary.thumbnail_bg_color, None)
         self.assertEqual(story_summary.thumbnail_filename, None)
+        with self.swap_to_always_return(
+            story_models.StorySummaryModel,
+            'get'
+            ):
+            story_summary = story_fetchers.get_story_summary_by_id('fakeID')
+            self.assertEqual(story_summary, None)
+
+    def test_get_completed_node_id(self):
+        self.assertEqual(
+            story_fetchers.get_completed_node_ids('randomID', 'someID'),
+            []
+        )
+        story_services.record_completed_node_in_story_context(
+            self.USER_ID, self.STORY_ID, self.NODE_ID_1)
+        story_services.record_completed_node_in_story_context(
+            self.USER_ID, self.STORY_ID, self.NODE_ID_2)
+        self.assertEqual(
+            story_fetchers.get_completed_node_ids(self.USER_ID, self.STORY_ID),
+            [self.NODE_ID_1, self.NODE_ID_2]
+        )
+
+    def test_get_pending_and_all_nodes_in_story(self):
+        result = story_fetchers.get_pending_and_all_nodes_in_story(
+            self.USER_ID, self.STORY_ID
+        )
+        pending_nodes = result['pending_nodes']
+        self.assertNotEqual(pending_nodes, [])
```

Comment:
Would it be possible to test something that is not empty here?

Reply:
Taken care of.